lib/query.js

```diff
@@ -2291,9 +2293,9 @@ Query.prototype.mongooseOptions = function(v) {
 
 Query.prototype._castConditions = function() {
   let sanitizeFilterOpt = undefined;
-  if (this.model != null && utils.hasUserDefinedProperty(this.model.db.options, 'sanitizeFilter')) {
+  if (this.model != null && this.model.db.options?.sanitizeFilter != null) {
```

Comment:
Can also be shortened to:
```suggestion
  if (this.model?.db.options?.sanitizeFilter != null) {
```